### PR TITLE
custom connection class

### PR DESCRIPTION
**Pull request by julyclyde**
dependency injection

### CODE REVIEW

File: rq/cli/cli.py [hunk 1]
```diff
@@ -37,12 +37,12 @@
                              help='Module containing RQ settings.')
 
 
-def connect(url, config=None):
+def connect(url, config=None, connection_class='redis.StrictRedis'):
     if url:
-        return StrictRedis.from_url(url)
+        return connection_class.from_url(url)
```
Comment (proofit404) [inline rq/cli/cli.py]: attribute error: str does not have attribute from_url.

Comment (julyclyde) [inline rq/cli/cli.py]: I should have tested it before PR.

**Comment by selwin** [comment]
May I know the rationale behind this change?

**Comment by julyclyde** [comment]
I want to use RedisCluster class instead.

**Comment by selwin** [comment]
@julyclyde alright, merged. Could you please open a PR to `gh-pages` branch documenting this change with an example on how to use this feature with RedisCluster? I think that will be really useful. I also recently got an inquiry about this https://github.com/ui/django-rq/issues/177

**Pull request merged**